packages/compiler/core/projector.ts

```diff
@@ -324,11 +329,21 @@ export function createProjector(
     const projectedType = projectType(prop.type);
     const projectedDecs = projectDecorators(prop.decorators);
 
-    const projectedProp = shallowClone(prop, {
+    const projectedProp: ModelProperty = shallowClone(prop, {
       type: projectedType,
       decorators: projectedDecs,
     });
 
+    if (prop.model) {
+      const parentModel = projectType(prop.model) as Model;
```

Comment:
Is this cast safe? IIRC, projecType can return `never`?

Reply:
I think this works right now because projecting the parent always result in getting the projection mid projection

Reply:
Could we maybe for now change those to have a `compilerAssert` and then we can revisit if this ever happen?

Reply:
Sound good.

Reply:
I agree that this should be safe.

Reply:
I see. I missed the explaanation. I don't feel strongly about adding the asserts, then, but it couldn't hurt.